packages/expo/hooks/lib/upload-source-maps.js

```diff
@@ -0,0 +1,54 @@
+const { upload } = require('bugsnag-sourcemaps')
+const { promisify } = require('util')
+const { tmpdir } = require('os')
+const { sep, join } = require('path')
+const { mkdtemp, writeFile } = require('fs')
+
+module.exports = async (
+  apiKey,
+  iosManifest,
+  iosBundle,
+  iosSourceMap,
+  androidManifest,
+  androidBundle,
+  androidSourceMap,
+  projectRoot
+) => {
+  const dir = await makeTmpDir()
+
+  const androidBundlePath = join(dir, 'android.bundle.js')
+  const androidSourceMapPath = join(dir, 'android.bundle.js.map')
+
+  const iosBundlePath = join(dir, 'ios.bundle.js')
+  const iosSourceMapPath = join(dir, 'ios.bundle.js.map')
+
+  await promisify(writeFile)(iosSourceMapPath, iosSourceMap, 'utf-8')
```

Comment:
More of a question than a critique, but is `promisify(f)` impure  in some way? Reading this, I wasn't certain if these two were equivalent:

```js
await promisify(writeFile)(iosSourceMapPath, iosSourceMap, 'utf-8')
await promisify(writeFile)(iosBundlePath, iosBundle, 'utf-8')
```

```js
const asyncWriteFile = promisify(writeFile)
await asyncWriteFile(iosSourceMapPath, iosSourceMap, 'utf-8')
await asyncWriteFile(iosBundlePath, iosBundle, 'utf-8')
```

Reply:
Those examples would be functionally equivalent. Want me to change it over so we're not repeatedly calling `promisify()` on the same function?

Reply:
Yeah, its a bit more readable to me 👍